utils/mpi_listener.hpp

```diff
@@ -0,0 +1,161 @@
+/*
```

Comment:
Didn't we already have this (and more than one version, too)?

Reply:
Nope, was previously used from GT, but is not provided in the GT installation/public API anymore.